src/arr.ts

```diff
@@ -208,56 +217,74 @@ function getRelevantArrInstances(mediaType: MediaType): string[] {
 	switch (mediaType) {
 		case MediaType.SEASON:
 		case MediaType.EPISODE:
-			return sonarr ?? [];
+			return sonarr;
 		case MediaType.MOVIE:
-			return radarr ?? [];
+			return radarr;
+		case MediaType.ANIME:
+		case MediaType.OTHER:
+			return [...sonarr, ...radarr];
 		default:
 			return [];
 	}
 }
 
-export async function scanAllArrsForExternalIds(
+export async function scanAllArrsForMedia(
 	searchee: Searchee,
 	mediaType: MediaType,
-): Promise<Result<ExternalIds, boolean>> {
+): Promise<Result<ParsedMedia, boolean>> {
 	const uArrLs = getRelevantArrInstances(mediaType);
+	if (uArrLs.length === 0) {
+		return resultOfErr(false);
+	}
 	try {
+		const title =
+			mediaType === MediaType.OTHER
+				? cleanseSeparators(
+						stripExtension(searchee.name)
+							.replace(RELEASE_GROUP_REGEX, "")
+							.replace(RESOLUTION_REGEX, "")
+							.replace(REPACK_PROPER_REGEX, ""),
+					)
+				: searchee.name;
+		let error = new Error(`No ids found for ${title} [${mediaType}]`);
 		for (const uArrL of uArrLs) {
-			const externalIds = await getExternalIdsFromArr(searchee, uArrL);
-			if (Object.values(externalIds).some(isTruthy)) {
-				logArrQueryResult(externalIds, searchee.name, mediaType);
-				return resultOf(externalIds);
+			const name =
+				mediaType === MediaType.OTHER &&
+				getRuntimeConfig().sonarr.includes(uArrL)
+					? `${title} S00E00` // Sonarr needs season or episode
+					: title;
+			const result = await getMediaFromArr(uArrL, name);
+			if (result.isErr()) {
+				error = result.unwrapErr();
+				continue;
+			}
+			const response = result.unwrap();
+			const ids = response.movie ?? response.series;
+			if (Object.values(ids).some(isTruthy)) {
+				const label = response.movie ? Label.RADARR : Label.SONARR;
+				logArrQueryResult(ids, name, label);
+				return resultOf(response);
 			}
 		}
-		throw new Error("fall through to catch");
+		throw error;
 	} catch (error) {
```

Comment:
I think it's possible it could be changed to just if logic, at some point while writing this fetch would throw but that's been refactored a couple of times since...

I'll defer to @ShanaryS to handle this.